Sources/Plasma/FeatureLib/pfMetalPipeline/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+set(pfMetalPipeline_SOURCES
+    plMetalDevice.cpp
+    plMetalDeviceRefs.cpp
+    plMetalMaterialShaderRef.cpp
+    plMetalPipeline.cpp
+    plMetalPipelineState.cpp
+    plMetalPlateManager.cpp
+    plMetalShader.cpp
+    plMetalFragmentShader.cpp
+    plMetalVertexShader.cpp
+    plMetalTextFont.cpp
+    plMetalEnumerate.mm
+    plMetalDevicePerformanceShaders.mm
+)
+
+set(pfMetalPipeline_HEADERS
+    plMetalDevice.h
+    plMetalDeviceRef.h
+    plMetalMaterialShaderRef.h
+    plMetalPipeline.h
+    plMetalPipelineState.h
+    plMetalPlateManager.h
+    plMetalShader.h
+    plMetalTextFont.h
+    plMetalFragmentShader.h
+    plMetalVertexShader.h
+    ShaderSrc/ShaderTypes.h
+    ShaderSrc/ShaderVertex.h
+    pfMetalPipelineCreatable.h
+)
+
+plasma_library(pfMetalPipeline SOURCES ${pfMetalPipeline_SOURCES} ${pfMetalPipeline_HEADERS})
+target_link_libraries(pfMetalPipeline
+    PUBLIC
+        CoreLib
+        pnNucleusInc
+        plPipeline
+        "-framework Metal"
+    PRIVATE
+        plStatusLog
+    INTERFACE
+        pnFactory
+)
+target_include_directories(pfMetalPipeline PUBLIC "metal-cpp")
```

Comment:
Having this problem right now with the builds.
```
In file included from /Users/runner/work/Plasma/Plasma/Sources/Tests/FeatureTests/pfPythonTest/test_cyMisc.cpp:48:
In file included from /Users/runner/work/Plasma/Plasma/Sources/Plasma/FeatureLib/inc/pfAllCreatables.h:65:
In file included from /Users/runner/work/Plasma/Plasma/Sources/Plasma/FeatureLib/pfMetalPipeline/pfMetalPipelineCreatable.h:46:
In file included from /Users/runner/work/Plasma/Plasma/Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalPipeline.h:47:
In file included from /Users/runner/work/Plasma/Plasma/Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalDevice.h:49:
/Users/runner/work/Plasma/Plasma/Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalDeviceRef.h:47:10: fatal error: 'Metal/Metal.hpp' file not found
#include <Metal/Metal.hpp>
```

The problem is the tests need this header. I thought declaring this header as public would force downstream code to import.

It looks like the DX pipeline is freely able to declare DX types without issue. And the DX pipeline doesn't include the DX header. Stuff like D3DFORMAT is used directly in the DX pipeline header with no forward declaration. So my assumption is that something different is happening to force the DX header imports to be accepted everywhere.